### PR TITLE
I18n

### LINKED ISSUE #2: Internationalization

### PR DESCRIPTION
**Pull request by anthonygego**
Fixes #2.  Only the frontend GUI and course/tasks are concerned for now.

Language files for the frontend are found in ``inginious/frontend/webapp/i18n``. Files for course/tasks are found in ``$i18n/*.mo`` in each course/task folder (the ``$`` prefix is used to prevent using a valid taskid). The default task language is used in the frontend selected language is not available in the task.

### CODE REVIEW

File: inginious/frontend/webapp/templates/index.html [hunk 1]
```diff
@@ -5,34 +5,30 @@
...
 
 $# Start content
     <div class="row">
         <div class="col-sm-5">
-            <p>
-                <h2>Hello! <small>Welcome on the INGInious platform.</small></h2>
-                INGInious allows to securely grade code made by students.
-            </p>
-            <p>Please log in to see the courses.</p>
+            $:_("<p><h2>Hello! <small>Welcome on the INGInious platform.</small></h2>INGInious allows to securely grade code made by students.</p><p>Please log in to see the courses.</p>")
```
Comment (GuillaumeDerval) [inline inginious/frontend/webapp/templates/index.html]: Same here

File: inginious/frontend/webapp/templates/maintenance.html [hunk 2]
```diff
@@ -86,11 +86,11 @@
         <div class="navbar navbar-default navbar-static-top" style="position:absolute; width:100%" role="navigation">
             <div class="container-fluid">
                 <div class="navbar-header" style="width: 100%">
-                    <p class="navbar-text">&copy; 2014-2017 Universit&eacute; catholique de Louvain</p>
+                    <p class="navbar-text">$:_("&copy; 2014-2017 Universit&eacute; catholique de Louvain")</p>
```
Comment (GuillaumeDerval) [inline inginious/frontend/webapp/templates/maintenance.html]: No need to translate

File: inginious/frontend/webapp/templates/course_admin/classroom_edit.html [hunk 3]
```diff
@@ -14,9 +14,9 @@
 $def NavbarF():
     <ol class="nav navbar-nav nav-breadcrumb">
         <li><a href="$get_homepath()/course/$course.get_id()">$course.get_name()</a></li>
-        <li><a href="$get_homepath()/admin/$course.get_id()" title="$:_('Administration')" data-toggle="tooltip" data-placement="bottom">
+        <li><a href="$get_homepath()/admin/$course.get_id()" title=$:_('"Administration"') data-toggle="tooltip" data-placement="bottom">
```
Comment (GuillaumeDerval) [inline inginious/frontend/webapp/templates/course_admin/classroom_edit.html]: Shouldn't it be `title="$:_('Administration')"` ?

File: inginious/frontend/webapp/templates/course_admin/classroom_edit.html [hunk 4]
```diff
@@ -14,9 +14,9 @@
...
             <i class="fa fa-user-secret"></i></a></li>
-        <li><a href="$get_homepath()/admin/$course.get_id()/aggregations" title="$:_('Classrooms')" data-toggle="tooltip" data-placement="bottom">
+        <li><a href="$get_homepath()/admin/$course.get_id()/aggregations" title=$:_('"Classrooms"') data-toggle="tooltip" data-placement="bottom">
```
Comment (GuillaumeDerval) [inline inginious/frontend/webapp/templates/course_admin/classroom_edit.html]: Same here

File: inginious/frontend/webapp/templates/course_admin/danger_zone.html [hunk 5]
```diff
@@ -107,7 +107,7 @@ <h3 class="panel-title">$:_("Backups")</h3>
                                             <i class="fa fa-recycle"></i>
                                         </button>
 
-                                        <a href="?download=$backup['file']" class="btn btn-default btn-xs" title="$:_('Download backup')"
+                                        <a href="?download=$backup['file']" class="btn btn-default btn-xs" title=$:_('"Download backup"')
```
Comment (GuillaumeDerval) [inline inginious/frontend/webapp/templates/course_admin/danger_zone.html]: Why not `"$:_('Download backup')"` ?

File: inginious/frontend/webapp/templates/course_admin/download.html [hunk 6]
```diff
@@ -14,7 +14,7 @@
 $def NavbarF():
     <ol class="nav navbar-nav nav-breadcrumb">
         <li><a href="$get_homepath()/course/$course.get_id()">$course.get_name()</a></li>
-        <li><a href="$get_homepath()/admin/$course.get_id()" title="$:_('Administration')" data-toggle="tooltip" data-placement="bottom">
+        <li><a href="$get_homepath()/admin/$course.get_id()" title=$:_('"Administration"') data-toggle="tooltip" data-placement="bottom">
```
Comment (GuillaumeDerval) [inline inginious/frontend/webapp/templates/course_admin/download.html]: Why not `"$:_('Administration')"` ?

File: inginious/frontend/webapp/templates/course_admin/replay.html [hunk 7]
```diff
@@ -14,7 +14,7 @@
 $def NavbarF():
     <ol class="nav navbar-nav nav-breadcrumb">
         <li><a href="$get_homepath()/course/$course.get_id()">$course.get_name()</a></li>
-        <li><a href="$get_homepath()/admin/$course.get_id()" title="$:_('Administration')" data-toggle="tooltip" data-placement="bottom">
+        <li><a href="$get_homepath()/admin/$course.get_id()" title=$:_('"Administration"') data-toggle="tooltip" data-placement="bottom">
```
Comment (GuillaumeDerval) [inline inginious/frontend/webapp/templates/course_admin/replay.html]: Why not `"$:_('Administration')"`?

File: inginious/frontend/webapp/templates/course_admin/settings.html [hunk 8]
```diff
@@ -12,7 +12,7 @@
 $def NavbarF():
     <ol class="nav navbar-nav nav-breadcrumb">
         <li><a href="$get_homepath()/course/$course.get_id()">$course.get_name()</a></li>
-        <li><a href="$get_homepath()/admin/$course.get_id()" title="$:_('Administration')" data-toggle="tooltip" data-placement="bottom">
+        <li><a href="$get_homepath()/admin/$course.get_id()" title=$:_('"Administration"') data-toggle="tooltip" data-placement="bottom">
```
Comment (GuillaumeDerval) [inline inginious/frontend/webapp/templates/course_admin/settings.html]: Why not `"$:_('Administration')"` ?

File: inginious/frontend/webapp/templates/course_admin/edit_tabs/basic.html [hunk 9]
```diff
@@ -33,148 +33,151 @@
...
                     <input type="radio" value="false" name="store_all"
                         $if stored_submissions > 0:
                             checked="checked"
-                    /> Only the last
+                    /> $:_("Only the last {} sumbmissions").split("{}")[0]
                 </label>
             </div>
             <div class="col-xs-offset-1 col-lg-offset-0 col-xs-11 col-lg-2"><input name="stored_submissions" class="form-control input-xs" style="height:20px;padding: 0 10px;" value="$(stored_submissions if stored_submissions > 0 else '')" placeholder="5" type='number'></div>
-            <div class="col-xs-offset-1 col-lg-offset-0 col-xs-11 col-lg-3"><label>submission(s)</label></div>
+            <div class="col-xs-offset-1 col-lg-offset-0 col-xs-11 col-lg-3"><label>$:_("Only the last {} sumbmissions").split("{}")[1]</label></div>
```
[the remark this answers is not in the document]
Comment (anthonygego) [reply inginious/frontend/webapp/templates/course_admin/edit_tabs/basic.html]: ok

File: inginious/frontend/webapp/templates/course_admin/edit_tabs/files.html [hunk 10]
```diff
@@ -31,7 +31,7 @@
             <tr data-x-path="$complete_name">
                 <td>
                     $for i in range(level+1):
-                        &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
+                        $:("&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;")
```
[the remark this answers is not in the document]
Comment (anthonygego) [reply inginious/frontend/webapp/templates/course_admin/edit_tabs/files.html]: ok

File: inginious/frontend/webapp/templates/maintenance.html (hunk 2 above)
Comment (anthonygego) [reply inginious/frontend/webapp/templates/maintenance.html]: ok

File: inginious/frontend/webapp/templates/course_admin/replay.html (hunk 7 above)
Comment (anthonygego) [reply inginious/frontend/webapp/templates/course_admin/replay.html]: web.py templates and pybabel...

**Comment by anthonygego** [comment]
After discussion we'll keep html tags temporarily in pot strings as there is no easy way using webpy templates to split paragraphs and add comment for the translators to know that those paragraphs are linked.

File: inginious/frontend/webapp/templates/course_admin/edit_tabs/basic.html [hunk 11]
```diff
@@ -33,148 +33,151 @@
...
             <div class="col-xs-12 col-lg-3">
                 <label>
                     <input type="radio" value="soft" name="submission_limit"
                         $if submission_limit['period'] > 0:
                             checked="checked"
-                    /> Soft limit:
+                    /> $:_("Soft limit:")
                 </label>
             </div>
+            $ text = _("{} submission(s) every {} hour(s)").split("{}")
```
[the remark this answers is not in the document]
Comment (anthonygego) [reply inginious/frontend/webapp/templates/course_admin/edit_tabs/basic.html]: I'll use tags and replace it with the appropriate html input field

**Comment by anthonygego** [comment]
Update done

**Pull request merged**